cpp/arcticdb/codec/encoded_field.hpp

```diff
@@ -307,9 +307,9 @@ struct EncodedFieldImpl : public EncodedField {
     [[nodiscard]] const EncodedBlock& shapes(size_t n) const {
         util::check(shapes_count_ != 0, "No shape allocated");
         if(!is_old_style_shapes())
-            return *reinterpret_cast<const EncodedBlock*>(&blocks_[0]);
+            return blocks()[n];
```

Comment:
I think this should be an `n` but it was previously a `0`, so this needs to be confirmed.

Reply:
This should be 0. In the old style shapes we had a block of shapes followed by a block of values. The new style has only a single block of shapes and it's always located at 0.

Reply:
Maybe we should assert that it's zero, since asking for a shape block > n if it's not old style is always wrong

Reply:
Thanks, I've changed the index to 0 and added an assert.